### PR TITLE
Refactor TimeTablePage and related files.

### PR DESCRIPTION
**Pull request by georgecabaccang**
#### Note worthy:
- Create divider for general use.
- Create timetable.dart.
- Create DaysOfTheWeek enum for consistent day values.
- Create ClassDetails class for consistent class details. (sub, teacher, etc..)
- Create Timetable class to encapsulate DaysOfTheWeek and ClassDetails for better typing. (Map<String, dynamic>? timetableData; -> late Timetable? timeTable;)
- Remove 'Timetable' key from timetable.json and expose the days directly.
- Refactor TimetableCard into a pure card widget and use CustomItemCard.
- Create CustomListView for more uniform general listviews.
- Create TimetableList and CustomListView.
- Configure TimeTableTabBar to be more reusable.
- Create StringValues interface with regards to new configuration of TimeTableTabBar.
- Add 'shortened' getter to DaysOfTheWeek with regards to new configuration of TimeTableTabBar.
- Remove Expanded widget in CustomListView to remove minor display exception/bug.

### CODE REVIEW

File: lib/pages/timetable/helpers/timetable.dart [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                return 'WED';
+            case DaysOfTheWeek.thursday:
+                return 'THU';
+            case DaysOfTheWeek.friday:
+                return 'FRI';
+            case DaysOfTheWeek.saturday:
+                return 'SAT';
+        }
+    }
+}
```
Comment (Kelv1nG) [inline lib/pages/timetable/helpers/timetable.dart:46]: change implementation of value / display string similar to enums in lib/enums

File: lib/pages/timetable/widget/timetable_list.dart [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import 'package:flutter/material.dart';
+import 'package:school_erp/pages/common_widgets/lists/custom_list_view.dart';
+import 'package:school_erp/pages/timetable/helpers/timetable.dart';
+import 'package:school_erp/pages/timetable/widget/timetable_card.dart';
+
+class TimeTableList<T> extends StatelessWidget {
+    final List<ClassDetails> classes;
```
Comment (Kelv1nG) [inline lib/pages/timetable/widget/timetable_list.dart:7]: can removed generic T as it is unused(?)

File: lib/pages/timetable/widget/timetable_tabbar.dart [hunk 3]
```diff
@@ -1,52 +1,54 @@
 import 'package:flutter/material.dart';
+import 'package:school_erp/interfaces/string_values.dart';
 import 'package:school_erp/theme/colors.dart';
 
-class TimeTableTabBar extends StatefulWidget {
-  final TabController controller;
+class TimeTableTabBar<T extends StringValues> extends StatefulWidget {
+    final TabController controller;
+    final List<T> tabs;
```
Comment (Kelv1nG) [inline lib/pages/timetable/widget/timetable_tabbar.dart:7]: will this be used other than the subject class schedule?
If no probably just use a concrete type for List<T> tabs

**Review by Kelv1nG — CHANGES_REQUESTED** [review]
please check/comment on requested changes

**Review by ejempic — APPROVED** [review]

File: lib/pages/timetable/widget/timetable_list.dart (hunk 2 above)
Comment (georgecabaccang) [reply lib/pages/timetable/widget/timetable_list.dart]: Resolved. My bad.
Changes here: [6c119de](https://github.com/Kelv1nG/iskulapp-flutter/pull/17/commits/6c119debf0bb73ec7b53637cd2eb8376845f3d28)

File: lib/pages/timetable/widget/timetable_tabbar.dart (hunk 3 above)
Comment (georgecabaccang) [reply lib/pages/timetable/widget/timetable_tabbar.dart]: Changed it into reusable TabBar.
Changes here: [091d2c2](https://github.com/Kelv1nG/iskulapp-flutter/pull/17/commits/091d2c2c74e18c68809de64b688a2b764d049d2f)

File: lib/pages/timetable/helpers/timetable.dart (hunk 1 above)
Comment (georgecabaccang) [reply lib/pages/timetable/helpers/timetable.dart]: Resolved. Thank you for pointing this out.
Changes here: [e4dec7c](https://github.com/Kelv1nG/iskulapp-flutter/pull/17/commits/e4dec7c2203b7b63a93d0a818bb8e557b25ff29f)

File: lib/pages/common_widgets/dividers/general_divider.dart [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        double leftSpace = leftDividerSpaceWidth != null
+            ? leftDividerSpaceWidth!
+            : (symmetricDividerSpaceWidth != null
+                ? symmetricDividerSpaceWidth! * screenWidth / 100
+                : 0);
+        double rightSpace = rightDividerSpaceWidth != null
+            ? rightDividerSpaceWidth!
+            : (symmetricDividerSpaceWidth != null
+                ? symmetricDividerSpaceWidth! * screenWidth / 100
+                : 0);
```
[the remark this answers is not in the document]
Comment (georgecabaccang) [reply lib/pages/common_widgets/dividers/general_divider.dart]: Resolved. Sorry for the eyesore.
Changes here: [0361b54](https://github.com/Kelv1nG/iskulapp-flutter/pull/17/commits/0361b54a8d4aeaac4f27629679c006a394e12b99)

**Pull request merged**